cli/cmd/encore/deploy.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"github.com/logrusorgru/aurora/v3"
+	"github.com/spf13/cobra"
+
+	"encr.dev/cli/cmd/encore/cmdutil"
+	"encr.dev/cli/internal/platform"
+	"encr.dev/pkg/appfile"
+)
+
+var (
+	appSlug string
+	envName string
+	commit  string
+	branch  string
+	format  string
+)
+
+var deployAppCmd = &cobra.Command{
+	Use:                   "deploy --commit COMMIT_SHA | --branch BRANCH_NAME",
+	Short:                 "Deploy an Encore app to a cloud environment",
+	DisableFlagsInUseLine: true,
+	Hidden:                true,
+	Run: func(c *cobra.Command, args []string) {
+		if commit != "" {
+			hb, err := hex.DecodeString(commit)
```

Comment:
Thoughts on using `git rev-parse` here? Even if we require the commit to be pushed it still seems like a nice convenience. Maybe not worth it?

Reply:
Yeah, maybe we should. Can't assume that git is installed or the command is run from within a local repository. But I suppose I could run it if --app is not set and git is available. On fail, I could parse the the revision specifier manually and submit it as a commit sha if it's a 20 byte hex, or a branch if it's not.

Reply:
Probably rename the flag to revision then?

Reply:
I guess that's fair, let's just keep it as-is and not spend more time on it